src/transformers/modeling_utils.py

```diff
@@ -2087,3 +2098,54 @@ def prune_layer(layer, index, dim=None):
         return prune_conv1d_layer(layer, index, dim=1 if dim is None else dim)
     else:
         raise ValueError("Can't prune layer of class {}".format(layer.__class__))
+
+
+def apply_chunking_to_forward(
```

Comment:
This function is written very generally (can handle an arbitary number of input tensors) so that it can be applied to every kind of layer. It basically just chunks the layers operation over one dimension. In Reformer it is tested that backward and forward yield the same result.

Reply:
I would make forward_fn the first arg, but don't feel strongly. Very cool.